marlowe-protocols/src/Network/Protocol/Query/Types.hs

```diff
@@ -1,401 +1,386 @@
+    TagBin l r -> "(" <> reqName l <> " AND " <> reqName r <> ")"
+
+instance OTelRequest req => OTelProtocol (Query req) where
+  protocolName _ = "query." <> reqTypeName (Proxy @req)
   messageAttributes = curry \case
-    (_, MsgRequest query) -> MessageAttributes
-      { messageType = "request/" <> queryName (tagFromQuery query)
-      , messageParameters = [toPrimitiveAttribute $ T.pack $ showsPrecQuery 0 query ""]
+    (_, MsgRequest req) -> MessageAttributes
+      { messageType = "request/" <> reqName (tagFromReq req)
+      , messageParameters = [toPrimitiveAttribute $ T.pack $ show req]
       }
-    (ServerAgency (TokNext _ tag), MsgReject err) -> MessageAttributes
-      { messageType = "request/" <> queryName tag <> "/reject"
-      , messageParameters = toPrimitiveAttribute . T.pack <$> [showsPrecErr 0 tag err ""]
-      }
-    (ServerAgency (TokNext _ tag), MsgNextPage result mDelimiter) -> MessageAttributes
-      { messageType = "request/" <> queryName tag <> "/next_page"
-      , messageParameters = toPrimitiveAttribute . T.pack <$> showsPrecResult 0 tag result "" : ((\d -> showsPrecDelimiter 0 tag d "")<$> maybeToList mDelimiter)
-      }
-    (ClientAgency (TokPage tag), MsgRequestNext delimiter) -> MessageAttributes
-      { messageType = "request/" <> queryName tag <> "/next"
-      , messageParameters = [toPrimitiveAttribute $ T.pack $ showsPrecDelimiter 0 tag delimiter ""]
+    (ServerAgency (TokRes tag), MsgRespond a) -> MessageAttributes
+      { messageType = "request/" <> reqName tag <> "/respond"
+      , messageParameters = toPrimitiveAttribute . T.pack <$> [showsPrecResult 0 tag a ""]
       }
     (_, MsgDone) -> MessageAttributes
       { messageType = "done"
       , messageParameters = []
       }
 
-class IsQuery query => ArbitraryQuery query where
-  arbitraryTag :: Gen (SomeTag query)
-  arbitraryQuery :: Tag query delimiter err results -> Gen (query delimiter err results)
-  arbitraryDelimiter :: Tag query delimiter err results -> Maybe (Gen delimiter)
-  arbitraryErr :: Tag query delimiter err results -> Maybe (Gen err)
-  arbitraryResults :: Tag query delimiter err results -> Gen results
-  shrinkQuery :: query delimiter err results -> [query delimiter err results]
-  shrinkErr :: Tag query delimiter err results -> err -> [err]
-  shrinkResults :: Tag query delimiter err results -> results -> [results]
-  shrinkDelimiter :: Tag query delimiter err results -> delimiter -> [delimiter]
-
-class IsQuery query => QueryVariations query where
-  tags :: NonEmpty (SomeTag query)
-  queryVariations :: Tag query delimiter err results -> NonEmpty (query delimiter err results)
-  delimiterVariations :: Tag query delimiter err results -> [delimiter]
-  errVariations :: Tag query delimiter err results -> [err]
-  resultsVariations :: Tag query delimiter err results -> NonEmpty results
-
-instance ArbitraryQuery query => ArbitraryMessage (Query query) where
+class Request req => ArbitraryRequest req where
```

Comment:
Why isn't this located in a separate package for test generators. Are these arbitrary instances needed by the protocol?